src/thunderbots/software/test/grsim_communication/pivot_primitive.cpp

```diff
@@ -0,0 +1,146 @@
+    // parameterized pivot points
+    PivotPrimitive primitive =
+        PivotPrimitive(1, Point(10, -10), Angle::ofRadians(-1 / 4 * M_PI), 10);
+
+    // place the robot in the first quadrant, should rotate CW
+    Robot robot = Robot(1, GetParam(), Vector(0, 0), Angle::ofRadians(0.0),
+                        AngularVelocity::ofRadians(0.0), Timestamp::fromSeconds(0));
+
+    auto motion_command = get_motion_command(robot, primitive);
+
+    // figure out if the tangential vector is clockwise, take a cross product of a 2d
+    // vector in 3d plane
+    // https://math.stackexchange.com/questions/74307/two-2d-vector-angle-clockwise-predicate
+
+    // this vector connects the robot to the pivot point, the tangential vectors
+    // that look at both directions will be perpendicular to this vector
+    Vector radial_vector = robot.position() - primitive.getPivotPoint();
+    Vector direction_vector =
+        motion_command.global_destination - primitive.getPivotPoint();
+
+    // if c is less than 0, then direction vector is on the clockwise side of radial
+    // vector, which means the robot is rotating clockwise
+    float c = radial_vector.x() * direction_vector.y() -
+              radial_vector.y() * direction_vector.x();
+    EXPECT_LE(c, 0);
+}
+
+// Test picking a counter-clockwise path
+TEST_P(GrsimCommandPrimitiveVisitorParameterizedTest,
+       visit_pivot_primtive_counter_clockwise)
+{
+    // asked to pivot to 3/4 pi above x axis, with radius of 10, from point (-10, 10)
+    // (chosen far so all values pivot ccw) which should result in a cw rotation on all
+    // parameterized pivot points
+    PivotPrimitive primitive =
+        PivotPrimitive(1, Point(-10, 10), Angle::ofRadians(3 / 4 * M_PI), 10);
+
+    // place the robot in the first quadrant, should rotate CW
+    Robot robot = Robot(1, GetParam(), Vector(0, 0), Angle::ofRadians(0.0),
+                        AngularVelocity::ofRadians(0.0), Timestamp::fromSeconds(0));
+
+    auto motion_command = get_motion_command(robot, primitive);
+
+    // figure out if the tangential vector is clockwise, take a cross product of a 2d
+    // vector in 3d plane
+    // https://math.stackexchange.com/questions/74307/two-2d-vector-angle-clockwise-predicate
+
+    // this vector connects the robot to the pivot point, the tangential vectors
+    // that look at both directions will be perpendicular to this vector
+    Vector radial_vector = robot.position() - primitive.getPivotPoint();
+    Vector direction_vector =
+        motion_command.global_destination - primitive.getPivotPoint();
+
+    // if c is greater than 0, then direction vector is on the counter-clockwise side of
+    // radial vector, which means the robot is rotating counter-clockwise
+    float c = radial_vector.x() * direction_vector.y() -
+              radial_vector.y() * direction_vector.x();
+    EXPECT_GE(c, 0);
+}
+
+INSTANTIATE_TEST_CASE_P(Positions, GrsimCommandPrimitiveVisitorParameterizedTest,
```

Comment:
This is _slick_. Took me a second to figure out what's going on here, but it totally makes sense. Can we add a quick comment here explaining what this line is doing? I think that should be enough to push people in the right direction when they're reading through this.

Reply:
added a comment